.github/workflows/run_tests.yml

```diff
@@ -65,26 +65,13 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-      - name: Cache python libs
-        uses: actions/cache@v4
-        if: matrix.extras == 'test'
-        with:
-          path: |
-            # Only cache a subset of libraries, ensuring cache size remains under 10GB. See GH dsgibbons#42
-            ${{ env.pythonLocation }}/**/site-packages/pyspark*
-            ${{ env.pythonLocation }}/**/site-packages/nvidia*
-            ${{ env.pythonLocation }}/**/site-packages/torch*
-            ${{ env.pythonLocation }}/**/site-packages/functorch*
-          key: ${{ runner.os }}-${{ matrix.python-version }}-${{ env.pythonLocation }}-${{ hashFiles('setup.py', 'pyproject.toml') }}-0
       - name: Install libomp (macOS)
         if: matrix.os == 'macos-latest'
         run: brew install libomp
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          # Use "eager" update strategy in case cached dependencies are outdated
-          # Using regular install NOT editable install: see GH #3020
-          pip install --upgrade --upgrade-strategy eager '.[${{ matrix.extras }},plots]'
+          python -m pip install --upgrade uv
+          uv pip install --system '.[${{ matrix.extras }},plots]'
```

Comment:
@connortann shall we retain the 

```
# Using regular install NOT editable install: see GH #3020
```
comment? I think it's still relevant, right?

Reply:
Ah yes, absolutely. I'll add that back in.